stlab/future.hpp

```diff
@@ -564,59 +572,59 @@ class future<T, detail::enable_if_copyable<T>> {
 
     future() = default;
 
-    bool valid() const { return static_cast<bool>(_p); }
+    bool valid() const { return static_cast<bool>(std::atomic_load(&_p)); }
```

Comment:
What is the purpose of all of the atomic loads? The future itself cannot be shared between threads so I don't believe this is necessary.